src/Simulator/IncrementalRedstoneSimulator/RedstoneComparatorHandler.h

```diff
@@ -54,7 +54,13 @@ class cRedstoneComparatorHandler : public cRedstoneHandler
 
 			virtual bool Item(cBlockEntity * a_BlockEntity) override
 			{
-				auto & Contents = static_cast<cBlockEntityWithItems *>(a_BlockEntity)->GetContents();
+				auto BlockEntityWithItems = dynamic_cast<cBlockEntityWithItems *>(a_BlockEntity);
```

Comment:
Add a comment above this block of code, explaining why it's here. Something as simple as `// Skip BlockEntitites that have no contents` will work.